### PR TITLE
Add MCC loss

### LINKED ISSUE #8784: Add Matthews Correlation Coefficient (MCC) Loss
**Is your feature request related to a problem? Please describe.**
Medical image segmentation models trained with overlap-based loss functions like Dice and Tversky are susceptible to class imbalance, particularly when the foreground region is small relative to the background. These losses only consider TP, FP, and FN from the confusion matrix, ignoring true negatives (TN), which means they do not penalize misclassifications of background pixels. This can lead to suboptimal segmentation in datasets where background dominates the image.

**Describe the solution you'd like**
Add a Matthews Correlation Coefficient (MCC) loss function to monai.losses. The MCC loss uses all four entries of the confusion matrix (TP, TN, FP, FN) and has been shown to be effective for class-imbalanced segmentation. The loss is computed as 1 - MCC where MCC = (TP * TN - FP * FN) / sqrt((TP+FP)(TP+FN)(TN+FP)(TN+FN)), using soft confusion matrix entries to make it differentiable.                                                                                                                                                              

This was proposed in:

> K. Abhishek and G. Hamarneh, "Matthews Correlation Coefficient Loss for Deep Convolutional Networks: Application to Skin Lesion Segmentation," IEEE ISBI, 2021, pp. 225-229. https://doi.org/10.1109/ISBI48211.2021.9433782
                                                                                                                                                                                          
The paper has been cited 75 times, and the loss has also been adopted by the Segmentation Models PyTorch (smp) library. An implementation following MONAI conventions (supporting sigmoid, softmax, to_onehot_y, include_background, batch, reduction, etc.) along with unit tests is ready to submit as a PR.

**Describe alternatives you've considered**
- Using Dice loss or Tversky loss with tuned alpha/beta weights to mitigate class imbalance. These help but still fundamentally ignore TN.                                              
- Using compound losses like DiceCE or DiceFocal. These combine distance-based and overlap-based losses but add hyperparameter complexity without directly addressing all four confusion matrix entries.

**Additional context**
- Paper: https://doi.org/10.1109/ISBI48211.2021.9433782                                                                                                                                 
- Original code: https://github.com/kakumarabhishek/MCC-Loss
- [Segmentation Models PyTorch](https://smp.readthedocs.io/en/latest/index.html) adoption: the loss is available in smp as [smp.losses.MCCLoss](https://smp.readthedocs.io/en/latest/losses.html#mccloss)

### PR DESCRIPTION
**Pull request by kakumarabhishek**
Add Matthews Correlation Coefficient (MCC) Lossa

Fixes #8784 .

### Description

Add the Matthews Correlation Coefficient (MCC) loss function to monai.losses. Unlike Dice and Tversky losses which only use TP, FP, and FN, the MCC loss considers all four entries of the confusion matrix (TP, TN, FP, FN), making it effective for class-imbalanced segmentation tasks. The loss was proposed in Abhishek & Hamarneh (IEEE ISBI 2021), has been cited 75 times, and has been adopted by Segmentation Models PyTorch (smp). The implementation follows MONAI conventions, supporting sigmoid, softmax, to_onehot_y, include_background, batch, and reduction parameters.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by ericspod — APPROVED** [review]
Looks good to me, thanks!

File: monai/losses/__init__.py [hunk 1]
```diff
@@ -35,6 +35,7 @@
 from .focal_loss import FocalLoss
 from .giou_loss import BoxGIoULoss, giou
 from .hausdorff_loss import HausdorffDTLoss, LogHausdorffDTLoss
+from .mcc_loss import MCCLoss
 from .image_dissimilarity import GlobalMutualInformationLoss, LocalNormalizedCrossCorrelationLoss
```
Comment (ericspod) [inline monai/losses/__init__.py]:
```suggestion
from .image_dissimilarity import GlobalMutualInformationLoss, LocalNormalizedCrossCorrelationLoss
from .mcc_loss import MCCLoss
```
This should resolve the isort issue.
